integration/app_integration_test.go

```diff
@@ -1092,6 +1111,29 @@ func (p *pack) createAppSession(t *testing.T, publicAddr, clusterName string) st
 	err = json.NewDecoder(resp.Body).Decode(&casResp)
 	require.NoError(t, err)
 
+	require.Eventually(t, func() bool {
```

Comment:
Instead of making this a part of `createAppSession`, I would honestly decouple it and make a separate method e.g. `ensureAuditEvent` or something that would accept the event it expects.

Reply:
I've isolated the search part, but the assertion is still inside this function since it is specific. I've made a function similar to the [`checkHTTPResponse`](https://github.com/gravitational/teleport/blob/master/lib/srv/app/server_test.go#L415).

Reply:
Yeah the assertion is fine, but I was mostly wondering why `ensureAuditEvent` needs to be called as a part of `createAppSession`? Instead of checking for the event in effectively every app access test, can we have a dedicated test that specifically tests for expected audit log events?

For example, it could do:

1. `createAppSession` to init app session.
2. `ensureAuditEvent` to check that session.start event was emitted.
3. Make a request to the app.
4. `ensureAuditeEvent` again to check that session.chunk event was emitted.

Would be easier to troubleshoot and catch potential issues. And you also wouldn't need to update "createAppSession" function signature to pass extra 2 arguments. And as another benefit, no other app access test would potentially block for 500ms waiting for events. WDYT?

Reply:
That makes sense. I added this audit event assertion on the `createAppSession` since there seem to be multiple "forms" of apps, like being accessed by a leaf node and WS, so the idea was to guarantee that all of those were producing the correct Session start events. But having a case-specific test for audit tests sounds better to me also. I'm just not sure if it would require all of those app configurations or just one is enough.

Anyway, I'm going to update the tests.